File: Herencia/Ejercicio3/Java/Persona.java
package Herencia.Ejercicio3.Java;

public class Persona {
    protected String nombre, apellido;
    int ci, celular;
    int diaNac, mesNac, añoNac;

    public Persona (String nom, String ap, int ci, int cel, int d, int m, int a) {
        this.nombre = nom;
        this.apellido = ap; 
        this.ci = ci;
        this.celular = cel;
        this.diaNac = d;
        this.mesNac = m;
        this.añoNac = a;
        }
    int getEdad() {
        int diaAct = 13;
        int mesAct = 5;
        int añoAct = 2025;
        int edad = añoAct - añoNac;
        if (mesNac > mesAct || (mesNac == mesAct && diaNac > diaAct)) {
            edad--;
        }
        return edad;
        
}
    void mostrar() {
        System.out.println("Nombre y Apellido:" + nombre +"" + apellido); 
        System.out.println("CI:" + ci);
        System.out.println("Celular: " + celular);
        System.out.println("Fecha de nacimiento: "+diaNac+"/"+mesNac+"/"+añoNac); 
        System.out.println("Edad:"+ getEdad());
}

    String getApe() { return apellido; } 
    String getNom() { return nombre; }

    void mayor25() {
        if(getEdad()>25){
            System.out.println(nombre + ""+ apellido + "tiene" + getEdad());
        }
    }
}

File: Herencia/Ejercicio3/Python/Persona.py

class Persona:   

    def __init__(self, nombre, apellido, ci, celular, diaNac, mesNac, anioNac):        
        self.nombre = nombre
        self.apellido = apellido
        self.ci = ci
        self.celular = celular
        self.diaNac = diaNac       
        self.mesNac = mesNac
        self.anioNac = anioNac
    
    def getEdad(self):
        diaAct, mesAct, anioAct = 13, 5, 2025
        edad = anioAct - self.anioNac
        if self.mesNac > mesAct or (self.mesNac == mesAct and self.diaNac > diaAct):
            edad -= 1
            return edad
    def mostrar(self):
        print(f"Nombre y Apellido: {self.nombre} {self.apellido}")
        print(f"CI: {self.ci}")
        print(f"Celular: {self.celular}")
        print(f"Fecha de nacimiento: {self.diaNac} /{self.mesNac} /{self.anioNac}")
        print(f"Edad: {self.getEdad()}")
            
    def getApe(self): return self.apellido
    def getNom(self): return self.nombre
        
    def mayor25(self):
        
        if self.getEdad() > 25:
            print(f"{self.nombre} {self.apellido} tiene {self.getEdad()}")

